### PR TITLE
Add sa command to support service accounts

### PR DESCRIPTION
**Pull request by vadmeste**
A new command is added to `mc admin user` command called svc
and that's to create new services accounts from existing users.

It is also possible to show credentials of the existing service
accounts.

`mc admin user svc generate` and `mc admin user svc show`

### CODE REVIEW

**Comment by vadmeste** [comment]
Not WIP anymore

File: cmd/admin-user-svc-generate.go [hunk 1]
```diff
@@ -0,0 +1,108 @@
+/*
+ * MinIO Client (C) 2018 MinIO, Inc.
```
Comment (harshavardhana) [inline cmd/admin-user-svc-generate.go]: 2020

File: cmd/admin-user-svc-generate.go [hunk 2]
```diff
@@ -0,0 +1,108 @@
...
+PARENTUSER:
+  The parent user.
+
+FLAGS:
+  {{range .VisibleFlags}}{{.}}
+  {{end}}
+
+EXAMPLES:
+  1. Add a new service account under the name of 'foobar' to MinIO server.
+     {{.Prompt}} {{.HelpName}} myminio foobar /tmp/file.json
```
Comment (harshavardhana) [inline cmd/admin-user-svc-generate.go]: can we indicate that its optional to pass `/tmp/file.json` ?

File: cmd/admin-user-svc-generate.go [hunk 3]
```diff
@@ -0,0 +1,108 @@
...
+	Name:   "generate",
+	Usage:  "generate a new service account",
+	Action: mainAdminUserSVCGenerate,
+	Before: setGlobalsFromContext,
+	Flags:  globalFlags,
+	CustomHelpTemplate: `NAME:
+  {{.HelpName}} - {{.Usage}}
+
+USAGE:
+  {{.HelpName}} TARGET PARENTUSER
```
Comment (harshavardhana) [inline cmd/admin-user-svc-generate.go]:
```suggestion
  {{.HelpName}} TARGET PARENTUSER [POLICY_FILE]
```

File: cmd/admin-user-svc-generate.go [hunk 4]
```diff
@@ -0,0 +1,108 @@
...
+
+// svcMessage container for content message structure
+type svcMessage struct {
+	AccessKey    string `json:"accessKey,omitempty"`
+	SecretKey    string `json:"secretKey,omitempty"`
+	SessionToken string `json:"sessionToken,omitempty"`
+}
+
+func (u svcMessage) String() string {
+	dot := console.Colorize("SVC", "‣ ")
```
Comment (harshavardhana) [inline cmd/admin-user-svc-generate.go]: SVC should be SA

File: cmd/admin-user-svc.go [hunk 5]
```diff
@@ -0,0 +1,48 @@
...
+package cmd
+
+import (
+	"github.com/fatih/color"
+	"github.com/minio/cli"
+	"github.com/minio/minio/pkg/console"
+)
+
+var adminUserSVCCmd = cli.Command{
+	Name:   "svc",
```
Comment (harshavardhana) [inline cmd/admin-user-svc.go]:
```suggestion
	Name:   "sa",
```

SA -> service account

**Comment by vadmeste** [comment]
Not WIP. SA command supports empty policy now

File: cmd/admin-user-sa-generate.go [hunk 6]
```diff
@@ -0,0 +1,120 @@
...
+USAGE:
+  {{.HelpName}} TARGET PARENT-USER [POLICY_FILE]
+
+PARENT-USER:
+  The parent user.
+
+POLICY_FILE:
+  The path of the policy to apply for the new service account.
+  When unspecified, the policy of the parent user will be evaluated
+  instead for all type of service accounts requests.
```
Comment (harshavardhana) [inline cmd/admin-user-sa-generate.go]:
```suggestion
  Optional path of the policy to apply for the new service account.
  When not specified, the policy of the parent user is inherited, please
  check 'mc admin user info <parent_user>' for the relevant policy.
```

File: cmd/admin-user-sa-generate.go [hunk 7]
```diff
@@ -0,0 +1,120 @@
...
+
+FLAGS:
+  {{range .VisibleFlags}}{{.}}
+  {{end}}
+
+EXAMPLES:
+  1. Add a new service account under the name of 'foobar' to MinIO server.
+     {{.Prompt}} {{.HelpName}} myminio foobar /tmp/policy.json
```
Comment (harshavardhana) [inline cmd/admin-user-sa-generate.go]: provide an example without the policy

Comment (vadmeste) [reply cmd/admin-user-sa-generate.go]: > provide an example without the policy

done, https://github.com/minio/mc/pull/3105/commits/a5f3b1d6fcfbb938c1879f9c2465f7b9a8d7e87f

**Comment by harshavardhana** [comment]
@vadmeste can you get this reviewed from @abperiasamy ?

**Comment by harshavardhana** [comment]
Closing this PR as this is deprecated and not going to be implemented, as this functionality completely moves to `mcs`

**Pull request closed without merging**